src/mappings/poolKeeper.ts

```diff
@@ -1,11 +1,19 @@
-import { UpkeepSuccess, KeeperPayment } from '../../generated/schema';
+import { UpkeepSuccess, KeeperPayment, Upkeep } from '../../generated/schema';
 import { UpkeepSuccessful, KeeperPaid } from '../../generated/templates/PoolKeeper/PoolKeeper';
 
 // event UpkeepSuccessful(address indexed pool, bytes data, int256 indexed startPrice, int256 indexed endPrice);
 export function upkeepSuccessful(event: UpkeepSuccessful): void {
 	let upkeepSuccessId = event.params.pool.toHexString()+"-"+event.block.number.toString();
 
   let upkeepSuccess = new UpkeepSuccess(upkeepSuccessId);
+  let upkeep = Upkeep.load(upkeepSuccessId);
+
+  if(upkeep) {
```

Comment:
is this meant to be !upkeep or intentionally if upkeep exists set new prices

Reply:
yep is intentional, should always exist because of the order events are emitted on chain but this check is for type safety